Emby.XmlTv/Emby.XmlTv/Classes/XmlTvReader.cs

```diff
@@ -495,16 +495,28 @@ public void ProcessEpisodeNum(XmlReader reader, XmlTvProgram result)
                     ParseMovieDbSystem(reader, result);
                     break;
                 case "SxxExx":
-                    // TODO
-                    // <episode-num system="SxxExx">S03E12</episode-num>
-                    reader.Skip();
+                    ParseSxxExxSystem(reader, result);
                     break;
                 default: // Handles empty string and nulls
                     reader.Skip();
                     break;
             }
         }
+        public void ParseSxxExxSystem(XmlReader reader, XmlTvProgram result)
+        {
+
 
+            string value = reader.ReadElementContentAsString().ToUpper();
+            //Validate format
+            if (Regex.IsMatch(value,"[S][0-9]+[E][0-9]+"))
+            {
+                //Split at E - take before and remove S
```

Comment:
if you're using a regex, why not extract fields it already matched? it would handle cases like `S001E005foo` easily if you just take groups from `S([0-9]+)E([0-9]+)`

Reply:
Good call.. was [sS] before I decided to ToUpper() the value and dodge the issue. Forgot to remove the brackets.

Reply:
`[sS]` is better than `ToUpper` imo
Also you should use `ToUpperInvariant` ;)

Reply:
Invariant change in place.

I'm no regex master in csharp so having trouble getting Regex.Match to behave as expected.

			string input = "s002e45b";
			Match res = Regex.Match(input, "[sS]([0-9]+)[eE]([0-9]+)");

Doesn't seem to work as expected. An along those lines I don't know how far we want to "stretch" the SxxExx system. I can't actually find any documentation that it exists as a standard anywhere. How loose do we want to get? We quickly run into the territory of supporting things like:

Se002-Ep032
s2/ep32
s002xe32

Perhaps it's best to just take some form of the current match and log any non complying entries as "malformed"?

Reply:
Got the Regex.Match to work. Not sure if res.Groups[1].Success and res.Groups[2].Success should be check before using value or if the res.Success logically covers both

Reply:
I don't know for sure but my wild guess is `res.Sucess` should cover both (doesn't make much sense otherwise TBH).